site/docs/hosting/vps.md

```diff
@@ -52,130 +167,616 @@ systemd is a powerful service manager which is pre-installed on many Linux distr
+# sure your domain's A/AAAA DNS records are properly pointed to
+# this machine's public IP, then replace ":80" below with your
+# domain name.
+
+:80 {
+  # Set this path to your site's directory.
+  root * /usr/share/caddy
+
+  # Enable the static file server.
+  file_server
+
+  # Another common task is to set up a reverse proxy:
+  # reverse_proxy localhost:8080
+
+  # Or serve a PHP site through php-fpm:
+  # php_fastcgi localhost:9000
+}
+
+# Refer to the Caddy docs for more information:
+# https://caddyserver.com/docs/caddyfile
 ```
 
-### Managing Apps
+For the bot to work, make the configuration look like this:
 
-#### Start
+```text
+<domain> {
+  reverse_proxy /<token> localhost:<port>
+}
+```
+
+> Replace `<domain>` with your domain, `<token>` with your bot token, and `<port>` with the port on which you want to run your bot.
+
+Reload Caddy every time you change the site's configuration file using the following command:
 
 ```sh
-pm2 start --name <app_name> <entry_point>
+systemctl reload caddy
 ```
 
-> The `<app_name>` can be any identifier to your app, for example: `bot1`.
-> The `<entry_point>` should be the path to your index file (which runs your bot).
+Now all requests to the address `https://<domain>/<token>` will be redirected to the address `http://localhost:<port>/<token>`, where the bot's webhook is running.
 
-#### Restart
+#### Connecting a Webhook to Telegram
 
-By restarting, you stop the app, and start it again.
+All you have to do is tell Telegram where to send the updates.
+To do this, open your browser and visit the page at the following link:
 
-```sh
-pm2 restart <app_name>
+```text
+https://api.telegram.org/bot<token>/setWebhook?url=https://<domain>/<token>
 ```
 
-#### Reload
+> Replace `<token>` with your bot token and `<domain>` with your domain.
+
+## CI/CD
 
-By reloading, you replace the current process of your app with a new one, resulting in a 0-second downtime.
-This is recommended for stateless applications.
+[CI/CD](https://about.gitlab.com/topics/ci-cd) is an important part of the modern software development process.
+This guide covers almost all of the [CI/CD pipeline](https://about.gitlab.com/topics/ci-cd/cicd-pipeline).
+
+We will focus on writing scripts for GitHub and GitLab.
+You can easily adapt the examples below to your CI/CD service of choice, such as Jenkins, Buddy, etc., if needed.
+
+:::tip Self-hosted Runner
+GitHub and GitLab offer a certain amount of resources for free to complete your tasks.
+However, when setting up your pipeline, you may quickly use them all up, which will require you to pay money for additional resources or the tasks will not run.
+To prevent this from happening, we recommend installing a self-hosted runner on your server so that tasks run on your server.
+This way, you'll get rid of the limits and be able to run pipelines of any complexity (everything will depend only on the capacity of your server).
+:::
```

Comment:
@niusia-ua we can also just remove it entirely. The guide is good enough to make a first iteration work. That is the most important thing.
```suggestion
```
I assume that many people won't even have this problem in the first days of development, and they can always fix things later.

Reply:
Okay, this is an acceptable decision